src/core/a-mixin.js

```diff
@@ -9,106 +9,97 @@ var MULTIPLE_COMPONENT_DELIMITER = '__';
-        this.isMixin = true;
-      }
-    },
+class AMixin extends ANode {
+  static get observedAttributes () {
+    return Object.keys(components);
+  }
 
-    attributeChangedCallback: {
-      value: function (attr, oldVal, newVal) {
-        this.cacheAttribute(attr, newVal);
-        this.updateEntities();
-      }
-    },
+  constructor (self) {
+    super(self);
+    self = self || this;
+    self.componentCache = {};
+    self.isMixin = true;
+  }
 
-    attachedCallback: {
-      value: function () {
-        this.sceneEl = this.closestScene();
-        this.cacheAttributes();
-        this.updateEntities();
-        this.load();
-      }
-    },
+  attributeChangedCallback (attr, oldVal, newVal) {
+    super.attributeChangedCallback(attr, oldVal, newVal);
+    this.cacheAttribute(attr, newVal);
+    this.updateEntities();
+  }
 
-    /**
-     * setAttribute that parses and caches component values.
-     */
-    setAttribute: {
-      value: function (attr, value) {
-        window.HTMLElement.prototype.setAttribute.call(this, attr, value);
-        this.cacheAttribute(attr, value);
-      }
-    },
+  connectedCallback () {
+    this.id = this.getAttribute('id');
+    this.sceneEl = this.closestScene();
+    this.cacheAttributes();
+    this.updateEntities();
+    this.load();
+  }
 
-    /**
-     * If `attr` is a component, then parse the value using the schema and store it.
-     */
-    cacheAttribute: {
-      value: function (attr, value) {
-        var component;
-        var componentName;
+  /**
+   * setAttribute that parses and caches component values.
+   */
+  setAttribute (attr, value) {
+    window.HTMLElement.prototype.setAttribute.call(this, attr, value);
+    this.cacheAttribute(attr, value);
+  }
 
-        // Get component data.
-        componentName = utils.split(attr, MULTIPLE_COMPONENT_DELIMITER)[0];
-        component = components[componentName];
-        if (!component) { return; }
-        if (value === undefined) {
-          value = window.HTMLElement.prototype.getAttribute.call(this, attr);
-        }
-        this.componentCache[attr] = component.parseAttrValueForCache(value);
-      }
-    },
+  /**
+   * If `attr` is a component, then parse the value using the schema and store it.
+   */
+  cacheAttribute (attr, value) {
+    var component;
+    var componentName;
 
-    /**
-     * If `attr` is a component, then grab pre-parsed value from the cache.
-     * Else do a normal getAttribute.
-     */
-    getAttribute: {
-      value: function (attr) {
-        return this.componentCache[attr] ||
-               window.HTMLElement.prototype.getAttribute.call(this, attr);
-      }
-    },
+    // Get component data.
+    componentName = utils.split(attr, MULTIPLE_COMPONENT_DELIMITER)[0];
+    component = components[componentName];
+    if (!component) { return; }
+    if (value === undefined) {
+      value = window.HTMLElement.prototype.getAttribute.call(this, attr);
```

Comment:
`super.setAttribute(attr, value)` should work here too.

Reply:
thanks. that's code predating this PR.

Reply:
similar to above. we don't want the a-node setAttribute logic to run but we still want the new values to be reflected in the DOM